src/libraries/System.Reflection/tests/TypeInfoTests.cs

```diff
@@ -1173,6 +1173,56 @@ public void MakePointerType(Type type)
             Assert.True(pointerType.IsPointer);
         }
 
+        public static IEnumerable<object[]> TypeDescTypesIsPublic_TestData()
```

Comment:
`TypeDesc` is internal implementation detail of CoreCLR runtime. This test name does not make sense on Mono or NativeAOT. 

Also, this is testing more than `TypeDesc`s - for example, `TI_BaseClass.InternalNestedGenericClass<string>` is not represented by `TypeDesc`.